File: app/src/main/java/com/example/mlallemant/mentalbattle/UI/Session/Fragment/LoadingFragment.java
package com.example.mlallemant.mentalbattle.UI.Session.Fragment;


import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.mlallemant.mentalbattle.R;
import com.example.mlallemant.mentalbattle.UI.Session.SessionActivity;
import com.example.mlallemant.mentalbattle.Utils.DatabaseManager;
import com.example.mlallemant.mentalbattle.Utils.Player;
import com.example.mlallemant.mentalbattle.Utils.Session;
import com.example.mlallemant.mentalbattle.Utils.Utils;


/**
 * Created by m.lallemant on 17/11/2017.
 */

public class LoadingFragment extends Fragment {

    //UTILS
    private DatabaseManager db;
    private Session session;
    private Player currentPlayer;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.loading_activity, container, false);

        Bundle bundle = getArguments();
        session = bundle.getParcelable("session");
        currentPlayer = bundle.getParcelable("currentPlayer");

        db = DatabaseManager.getInstance();
        db.initListenerCurrentSession(session);

        initListener();
        return v;
    }

    private void initListener() {
        db.setOnSessionUpdateListener(new DatabaseManager.OnSessionUpdateListener() {
            @Override
            public void updateSessionUI(Session session) {
                if (session != null) {

                    boolean isEverybodyReady = true;

                    for (Player player : session.getPlayerList()) {
                        if (player.getReady().equals(Utils.SESSION_RDY_NO)) {
                            isEverybodyReady = false;
                        }
                    }

                    if (isEverybodyReady) {
                        launchTransitionFragment(session);
                    }
                }
            }
        });
    }

    private void launchTransitionFragment(Session session){
        SessionActivity sa = (SessionActivity) getActivity();
        sa.addCurrentRoundSessionNumber();

        db.removeListenerCurrentSession(session);
        TransitionFragment tf = new TransitionFragment();

        Bundle args = new Bundle();
        args.putInt("currentRoundSessionNumber", sa.getCurrentRoundSessionNumber());
        args.putParcelable("session", session);
        args.putParcelable("currentPlayer", currentPlayer);
        tf.setArguments(args);

        FragmentManager fm = getFragmentManager();
        FragmentTransaction ft = fm.beginTransaction();
        ft.replace(R.id.fl_session, tf);
        ft.commit();
    }

}

File: app/src/main/java/com/example/mlallemant/mentalbattle/ui/session/fragment/LoadingFragment.kt
package com.example.mlallemant.mentalbattle.ui.session.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.mlallemant.mentalbattle.R
import com.example.mlallemant.mentalbattle.ui.session.SessionActivity
import com.example.mlallemant.mentalbattle.utils.DatabaseManager
import com.example.mlallemant.mentalbattle.utils.Player
import com.example.mlallemant.mentalbattle.utils.Session
import com.example.mlallemant.mentalbattle.utils.Utils

/**
 * Created by m.lallemant on 17/11/2017.
 */
class LoadingFragment : Fragment() {

    //UTILS
    private var db: DatabaseManager? = null
    private var currentPlayer: Player? = null

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val v = inflater.inflate(R.layout.loading_activity, container, false)
        val bundle = arguments
        val session: Session =
            bundle!!.getParcelable("session")!!

        currentPlayer = bundle.getParcelable("currentPlayer")
        db = DatabaseManager.getInstance()
        db?.initListenerCurrentSession(session)
        initListener()
        return v
    }

    private fun initListener() {
        db!!.setOnSessionUpdateListener { session ->
            if (session != null) {
                var isEverybodyReady = true
                for (player in session.playerList) {
                    if (player.ready == Utils.SESSION_RDY_NO) {
                        isEverybodyReady = false
                        break
                    }
                }
                if (isEverybodyReady) {
                    launchTransitionFragment(session)
                }
            }
        }
    }

    private fun launchTransitionFragment(session: Session) {
        val sa = activity as SessionActivity?
        sa!!.addCurrentRoundSessionNumber()
        db!!.removeListenerCurrentSession(session)
        val tf = TransitionFragment()
        val args = Bundle()
        args.putInt("currentRoundSessionNumber", sa.currentRoundSessionNumber)
        args.putParcelable("session", session)
        args.putParcelable("currentPlayer", currentPlayer)
        tf.arguments = args
        val fm = fragmentManager
        val ft = fm!!.beginTransaction()
        ft.replace(R.id.fl_session, tf)
        ft.commit()
    }
}